cpp/src/arrow/compute/exec/aggregate.h

```diff
@@ -37,6 +37,7 @@ namespace internal {
 /// and use an aggregate node.
```

Comment:
I think the comment ending here is not right because `GroupBy` is also used by `pyarrow._compute._group_by`, which is also exposed in `pyarrow.compute`.

Reply:
The comment is right I think.  This wasn't really meant to be exposed externally.  This approach only works if you have all of your data in memory.  So there is no way to hook up `GroupBy` with flight or a dataset for example.  I didn't actually notice it was being used by python and given that this is already released I suppose it is something we will be living with anyways.

That being said, for the development of an ordered group-by algorithm, it seems like using this should be ok.

Reply:
I'll leave the comment as is, then.